### PR TITLE
[WIP] Redesign Header

### PR DESCRIPTION
**Pull request by ivarnakken**
# ⚠️ Under construction 👷🏼‍♂️ 

My proposition for a ✨redesigned✨ Header

**TODO**
- [x] New dropdown functionality
- [x] Improve visuals
- [ ] Register dropdown
- [ ] Mobile support
- [ ] Dark-theme check

---

Closes ABA-143

### CODE REVIEW

File: app/components/Header/DropdownContainer/FadeContents.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// @flow
+
+import React, { forwardRef } from 'react';
+import type { Node } from 'react';
+import styled, { keyframes } from 'styled-components';
```
Comment (LudvigHz) [inline app/components/Header/DropdownContainer/FadeContents.js:5]: Why is there styled-components in here?

File: app/components/Header/DropdownContainer/FadeContents.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+// @flow
+
+import React, { forwardRef } from 'react';
```
Comment (LudvigHz) [inline app/components/Header/DropdownContainer/FadeContents.js:3]: Remember to remove the React import from all files. We don't need them anymore

File: app/routes/events/components/EventDetail/index.js [hunk 3]
```diff
@@ -277,7 +277,7 @@ export default class EventDetail extends Component<Props> {
             )}
             {event.title}
           </ContentHeader>
-
+          {''}
```
Comment (SmithPeder) [inline app/routes/events/components/EventDetail/index.js:280]: hmm

File: app/routes/users/components/Penalties.js [hunk 4]
```diff
@@ -56,6 +56,7 @@ function Penalties({
                   <ConfirmModalWithParent
                     title="Slett prikk"
                     message="Er du sikker på at du vil slette denne prikken?"
+                    closeOnConfirm
```
Comment (SmithPeder) [inline app/routes/users/components/Penalties.js:59]: Cool, but why is this in this PR?

Comment (ivarnakken) [reply app/routes/users/components/Penalties.js:59]: Good question 🤔

File: app/components/Header/DropdownContainer/FadeContents.js (hunk 1 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContainer/FadeContents.js:5]: It made life a whole lot easier, but I plan on removing it in the end, it's just a low priority right now

File: app/components/Header/DropdownContainer/FadeContents.js (hunk 2 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContainer/FadeContents.js:3]: Good point!

File: app/components/Header/DropdownContainer/FadeContents.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  return keyframes`
+  to {
+    transform: translateX(0px);
+    opacity: ${animatingOut ? 0 : 1};
+  }
+`;
+};
+
+const FadeContainer = styled.div`
+  animation-name: ${getFadeContainerKeyFrame};
```
Comment (SmithPeder) [inline app/components/Header/DropdownContainer/FadeContents.js:19]: How does this work? Calling `getFadeContainerKeyFrame` without `{direction}` will `return;` so what is the point here? Is the `${getFadeContainerKeyFrame}` automatically called the prop values that match
```
      animatingOut={animatingOut}
      direction={direction}
```

Comment (SmithPeder) [reply app/components/Header/DropdownContainer/FadeContents.js:19]: Because in the next line you use the props directly, which is a behavior I understand.
```
  opacity: ${(props) => (props.direction && !props.animatingOut ? 0 : 1)};
  ```

File: app/components/Header/DropdownContainer/index.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+};
+
+const getFirstDropdownSectionHeight = (el) => {
+  if (
+    !el ||
+    !el.querySelector ||
+    !el.querySelector('*[data-first-dropdown-section]')
+  )
+    return 0;
+  return el.querySelector('*[data-first-dropdown-section]').offsetHeight;
```
Comment (SmithPeder) [inline app/components/Header/DropdownContainer/index.js:24]: This should be a `try catch`, i think that looks a bit more clear
```js
try {
  return el.querySelector('*[data-first-dropdown-section]').offsetHeight
} catch {
  return 0
}
```

File: app/components/Header/DropdownContainer/FadeContents.js (hunk 5 above)
Comment (SmithPeder) [reply app/components/Header/DropdownContainer/FadeContents.js:19]: I see that more code have this same behavior 🙃 , so I guess that's how it works. I kinda h8 that thooo, but okey 🥇 

One has to read three things in order to understand what values are being used:
1) This line `animation-name: ${getFadeContainerKeyFrame};`
2) This line `const getFadeContainerKeyFrame = ({ animatingOut, direction }) => {`
3) These lines
```
    <FadeContainer
      className={styles.fadeContainer}
      // prevent screen readers from reading out hidden content
      aria-hidden={animatingOut}
      animatingOut={animatingOut}
      direction={direction}
      ref={ref}
    >
 ```

Who figured that would be a good design pattern?

File: app/components/Header/DropdownContents/EventsDropdown.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
+// @flow
+
+import React from 'react';
+import Icon from '../../Icon';
+import styles from '../Header.css';
+
+const EventsDropdown = () => {
+  return (
```
Comment (SmithPeder) [inline app/components/Header/DropdownContents/EventsDropdown.js:8]: implicit return

File: app/components/Header/DropdownContents/EventsDropdown.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+            <h4>Kalender</h4>
+            <Icon className={styles.icon} name="arrow-forward" size={20} />
+          </div>
+        </a>
+      </div>
+      <div className={styles.dropdownSection}>
+        <a className={styles.altText} href="/users/me">
+          Dine arrangementer
+        </a>
+      </div>
```
Comment (SmithPeder) [inline app/components/Header/DropdownContents/EventsDropdown.js:34]: There are a bunch of `<a>` tags here. That will wreck the SPA. Use `<Link>` from react-router

File: app/components/Header/DropdownContents/InfoDropdown.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import styles from '../Header.css';
+
+const InfoDropdown = () => {
+  return (
```
Comment (SmithPeder) [inline app/components/Header/DropdownContents/InfoDropdown.js:5]: implicit return

File: app/components/Header/DropdownContents/InfoDropdown.js [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+            <p>Abakus</p>
+          </div>
+        </div>
+      </div>
+      <div className={styles.dropdownSection}>
+        <a className={styles.altText} href="/contact">
+          Kontakt oss
+        </a>
+      </div>
+    </div>
```
Comment (SmithPeder) [inline app/components/Header/DropdownContents/InfoDropdown.js:44]: That's a lot of `divs`. I haven't compared the `css`, but do `styles.dropdownSection` really demand `<div><div>` for each child element?

File: app/components/Header/DropdownContents/InfoDropdown.js [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+            <p>undergrupper er</p>
+          </div>
+          <div>
+            <h4>AbakusRevyen</h4>
+            <p>Abakus</p>
+          </div>
+        </div>
+      </div>
+      <div className={styles.dropdownSection}>
+        <a className={styles.altText} href="/contact">
```
Comment (SmithPeder) [inline app/components/Header/DropdownContents/InfoDropdown.js:40]: <a>

File: app/components/Header/DropdownContents/InfoDropdown.js (hunk 10 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContents/InfoDropdown.js:44]: No, it really doesn't, I'm just experimenting ;p

File: app/components/Header/DropdownContents/ProfileDropdown.js [hunk 12]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import styles from '../Header.css';
+
+const ProfileDropdown = () => {
+  return (
```
Comment (SmithPeder) [inline app/components/Header/DropdownContents/ProfileDropdown.js:5]: Implicit return

File: app/components/Header/Header.css [hunk 13]
```diff
@@ -120,50 +120,211 @@
...
-  &.activeItem {
-    color: var(--color-black);
-  }
+.dropdownRoot {
+  transform-origin: 0 0;
+  will-change: transform;
+  animation-duration: 300ms;
+  /* use 'forwards' to prevent flicker on leave animation */
+  animation-fill-mode: forwards;
+  /* flex styles will center the caret child component */
```
Comment (SmithPeder) [inline app/components/Header/Header.css:277]: image

File: app/components/Header/Navbar/index.js [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+  render() {
+    const { children, onMouseLeave } = this.props;
+    return (
+      <div className={styles.navbarEl} onMouseLeave={onMouseLeave}>
+        <div className={styles.navbarList}>{children}</div>
+      </div>
+    );
+  }
+}
+
```
Comment (SmithPeder) [inline app/components/Header/Navbar/index.js:22]: `const Navbar  = () =>`

File: app/components/Header/Navbar/NavbarItem.js [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+  title: any,
+  to: string,
+  index: number,
+  children: Node,
+};
+
+export default class NavbarItem extends Component<Props> {
+  onMouseEnter = () => {
+    this.props.onMouseEnter(this.props.index);
+  };
```
Comment (SmithPeder) [inline app/components/Header/Navbar/NavbarItem.js:20]: `const NavBaritem  = () =>`

File: app/components/Icon/Icon.css [hunk 16]
```diff
@@ -1,12 +1,16 @@
...
   color: var(--color-white);
-  border-radius: 8px;
-  min-width: 28px;
+  border-radius: 50%;
+  width: 28px;
+  height: 28px;
   font-size: 16px;
   padding: 5px;
-  opacity: 0.9;
 }
```
Comment (SmithPeder) [inline app/components/Icon/Icon.css:16]: I'm guessing this is for the profile page or something? Can this be in another PR?

File: app/components/Header/DropdownContents/NotificationsDropdown.js [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+                wordsAgo
+                style={{ margin: '0', display: 'block' }}
+              />
+            </div>
+          </div>
+        </div>
+      </Link>
+    );
+  }
+  return null;
```
Comment (SmithPeder) [inline app/components/Header/DropdownContents/NotificationsDropdown.js:49]: Negate this with
```
if (!renders) return;
return (
...
)
```

File: app/routes/events/components/EventDetail/index.js (hunk 3 above)
Comment (ivarnakken) [reply app/routes/events/components/EventDetail/index.js:280]: 👀

File: app/components/Icon/Icon.css (hunk 16 above)
Comment (ivarnakken) [reply app/components/Icon/Icon.css:16]: This is part of the badge you get on the notification icon, in the navbar, which makes it rather fitting for this PR 😄 To be fair though, I wasn't sure at first either ...

Comment (SmithPeder) [reply app/components/Icon/Icon.css:16]: Ahh, makes sence!

File: app/components/Header/DropdownContents/NotificationsDropdown.js (hunk 17 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContents/NotificationsDropdown.js:49]: I totally agree, even though it is a direct copy from our code base 🙃

Comment (SmithPeder) [reply app/components/Header/DropdownContents/NotificationsDropdown.js:49]: I bet a lot of my comments are from copy-pasted code but... 
> When you find a piece of thrash while cleaning out your old apartment do you move it to your new apartment?

Comment (ivarnakken) [reply app/components/Header/DropdownContents/NotificationsDropdown.js:49]: I know I know, just thinking of the implications here

Comment (SmithPeder) [reply app/components/Header/DropdownContents/NotificationsDropdown.js:49]: image

File: app/components/Header/DropdownContents/EventsDropdown.js (hunk 8 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContents/EventsDropdown.js:34]: This is also quite experimental, but ye, totally agree!

File: app/components/Header/DropdownContainer/index.js (hunk 6 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContainer/index.js:24]: Ye it made things a lot clearer 👍🏼

File: app/components/Header/DropdownContents/EventsDropdown.js (hunk 8 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContents/EventsDropdown.js:34]: This was also experimental, but agree! 😄

File: app/components/Header/DropdownContents/InfoDropdown.js (hunk 11 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContents/InfoDropdown.js:40]: Are you referring to how it should rather be a `<Link />`?

Comment (SmithPeder) [reply app/components/Header/DropdownContents/InfoDropdown.js]: Yeah, it's an internal /contact link. So it should be handled by the router

**Comment by SmithPeder** [comment]
How does this look? Like, visually? I have some complaints (probably)

**Comment by ivarnakken** [comment]
> How does this look? Like, visually? I have some complaints (probably)

As of now, only core functionality has been implemented. Visuals have been mostly experimental up until now (as you may know by now). However, the next phase of the project is visual work, and I assume I'll push something pleasing by new years. Besides, If you're very curious, you can always `git pull` and check it out yourself ;)

**Comment by SmithPeder** [comment]
> Besides, If you're very curious, you can always git pull and check it out yourself ;)

giT pUlL aNd CheCk fOr iT oUt yOUrseLF

image

**Comment by SmithPeder** [comment]
`Module not found: Error: Can't resolve 'react-flip-toolkit' in '/Users/smith/code/abakus/lego-webapp/app/components/Header'`

**Comment by ivarnakken** [comment]
> `Module not found: Error: Can't resolve 'react-flip-toolkit' in '/Users/smith/code/abakus/lego-webapp/app/components/Header'`

`yarn add react-flip-toolkit` 👀

**Comment by SmithPeder** [comment]
hOw AboUT addING iT to THe `package.json` :yourfired:

**Comment by ivarnakken** [comment]
> hOw AboUT addING iT to THe `package.json` :yourfired:

I thought people would manage 🙃 Nice to see some interest in my PR though ❤️

**Comment by SmithPeder** [comment]
```
Module not found: Error: Can't resolve 'styled-components' in '/Users/smith/code/abakus/lego-webapp/app/components/Header/DropdownContainer
```

I see, so basically, I have to add one-and-one dependency, since you haven't added them to the `package.json`. Great work, I love it.

File: app/components/Header/Header.css [hunk 18]
```diff
@@ -113,61 +113,223 @@
...
+.navbarItemEl {
+  position: relative;
+  width: max-content;
+}
+
+.navbarItemTitle {
+  height: 2rem; /* TODO: optimize */
+  background: transparent;
+  border: 0;
+  font-family: -apple-system, system-ui, 'Segoe UI', 'Roboto', 'Helvetica Neue',
```
Comment (SmithPeder) [inline app/components/Header/Header.css:143]: The audacity to add a new font 😱  
image

... and not just any font
> Segoe UI — Segoe UI ("User Interface") is a member of the Segoe family used in Microsoft products for user interface text

@LudvigHz gonna be mad

File: app/components/Header/Header.css [hunk 19]
```diff
@@ -113,61 +113,223 @@
...
+  animation-duration: 300ms;
+  animation-fill-mode: forwards;
+  top: 0;
+  left: 0;
 }
 
-.logoLightMode,
-.logoDarkMode {
+.logoDarkMode,
+.logoLightMode {
```
Comment (SmithPeder) [inline app/components/Header/Header.css:332]: Good change

File: app/components/Header/DropdownContents/ProfileDropdown.js [hunk 20]
```diff
@@ -0,0 +1,18 @@
+// @flow
+
+import styles from '../Header.css';
+
+const ProfileDropdown = () => (
+  <div className={styles.profileDropdownEl}>
+    <div className={styles.dropdownSection} data-first-dropdown-section>
+      <div>Ola Nordmann</div>
+      <div>Innstillinger</div>
+      <div>Møteinnkallinger</div>
```
Comment (SmithPeder) [inline app/components/Header/DropdownContents/ProfileDropdown.js:10]: I'm guessing this is WIP? Or at least the Ola Nordmann Part

Comment (ivarnakken) [reply app/components/Header/DropdownContents/ProfileDropdown.js:10]: Yes. I needed to make a bunch of dummy components in order to test out the core functionality ;)

File: app/components/Header/DropdownContainer/index.js [hunk 21]
```diff
@@ -0,0 +1,139 @@
...
+  currentDropdownEl = createRef<any>();
+  prevDropdownEl = createRef<any>();
+
+  componentDidMount() {
+    updateAltBackground({
+      altBackground: this.altBackgroundEl,
+      prevDropdown: this.prevDropdownEl.current,
+      currentDropdown: this.currentDropdownEl.current,
+    });
+  }
```
Comment (LudvigHz) [inline app/components/Header/DropdownContainer/index.js:83]: Preferably, don't create new class components. Use `useRef`, `useEffect` here instead.

File: app/components/Header/DropdownContents/EventsDropdown/EventsDropdown.css [hunk 22]
```diff
@@ -0,0 +1,22 @@
+.eventsDropdownEl {
+  width: 20rem;
+}
+
+.dropdownLink {
+  display: flex;
+  flex-direction: row;
+  justify-content: space-evenly;
```
Comment (LudvigHz) [inline app/components/Header/DropdownContents/EventsDropdown/EventsDropdown.css:8]: This looks really wonky to me IMO. Don't know how others feel, but the fact that the icons and text are misaligned is really weird. Maybe use grid for this one? That way we can have the icons and link text column-aligned.

File: app/components/Header/DropdownContents/EventsDropdown/index.js [hunk 23]
```diff
@@ -0,0 +1,33 @@
...
+          prefix="ion-md-"
+          size={25}
+        />
+        <span>Kalender</span>
+      </Link>
+    </div>
+    <div className={shared.dropdownSection}>
+      <Link className={shared.bottomLink} to="/users/me">
+        Dine arrangementer
+      </Link>
```
Comment (LudvigHz) [inline app/components/Header/DropdownContents/EventsDropdown/index.js:28]: Why is there a span in the one space but not the other?

File: app/components/Header/DropdownContents/InfoDropdown/index.js [hunk 24]
```diff
@@ -0,0 +1,66 @@
...
+      className={cx(shared.dropdownSection, styles.gridSection)}
+      data-first-dropdown-section
+    >
+      <Link to="pages/info-om-abakus">
+        <div>
+          <span>Generelt</span>
+          <Icon className={styles.icon} name="arrow-forward" size={20} />
+        </div>
+        <p>Litt av hvert om Abakus</p>
+      </Link>
```
Comment (LudvigHz) [inline app/components/Header/DropdownContents/InfoDropdown/index.js:21]: Please dear lord separate this into it's own component...

File: app/components/Header/DropdownContents/LoginDropdown.js [hunk 25]
```diff
@@ -0,0 +1,69 @@
...
+    case 'forgotPassword':
+      title = 'Glemt passord';
+      form = <ForgotPasswordForm />;
+      break;
+    default:
+      break;
+  }
+
+  return (
+    <div className={styles.eventsDropdownEl}>
```
Comment (LudvigHz) [inline app/components/Header/DropdownContents/LoginDropdown.js:42]: `eventsDropdownEl`???

File: app/components/Header/DropdownContents/LoginDropdown.js [hunk 26]
```diff
@@ -0,0 +1,69 @@
...
+      <div className={styles.dropdownSection} data-first-dropdown-section>
+        {title}
+        <button
+          onClick={this.toggleForgotPassword}
+          className={styles.toggleButton}
+        >
+          Glemt passord
+        </button>
+        <span className={styles.toggleButton}>&bull;</span>
```
Comment (LudvigHz) [inline app/components/Header/DropdownContents/LoginDropdown.js:51]: What is `&bull;`?

File: app/components/Header/DropdownContents/NotificationsDropdown/index.js [hunk 27]
```diff
@@ -0,0 +1,91 @@
...
+type Props = {
+  notificationsData: Object,
+  fetchNotifications: () => void,
+  notifications: Array<Object>,
+  markAllNotifications: () => Promise<void>,
+};
+
+type State = {
+  alreadyFetchedNotifications: boolean,
+};
```
Comment (LudvigHz) [inline app/components/Header/DropdownContents/NotificationsDropdown/index.js:21]: Can't see this used anywhere?

File: app/components/Header/DropdownContents/ProfileDropdown/ProfileDropdown.css [hunk 28]
```diff
@@ -0,0 +1,42 @@
+@import url('https://fonts.googleapis.com/css2?family=Lora&display=swap');
```
Comment (LudvigHz) [inline app/components/Header/DropdownContents/ProfileDropdown/ProfileDropdown.css:1]: Do you really need a seperate font just for this... It just adds to the page load time with _another_ font being loaded, and for a single word.

File: app/components/Header/DropdownContents/ProfileDropdown/ProfileDropdown.css [hunk 29]
```diff
@@ -0,0 +1,42 @@
...
+  border-radius: 0.5em;
+  color: var(--lego-red);
+  font-family: 'Roboto', sans-serif;
+  font-size: 1.1em;
+  transition: background-color 100ms ease-in-out;
+}
+
+.profileDropdownEl .dropdownLink:hover,
+.profileDropdownEl .activeDropdownLink {
+  background-color: rgba(214, 212, 212, 0.3);
```
Comment (LudvigHz) [inline app/components/Header/DropdownContents/ProfileDropdown/ProfileDropdown.css:37]: Use variables.

File: app/components/Header/DropdownContents/ProfileDropdown/index.js [hunk 30]
```diff
@@ -0,0 +1,45 @@
...
+      >
+        <Icon name="calendar" size={20} />
+        <span>Møteinnkallinger</span>
+      </NavLink>
+    </div>
+    <div className={shared.dropdownSection}>
+      <span className={shared.bottomLink} onClick={logout}>
+        Logg ut
+        <Icon name="log-out" size={24} />
+      </span>
```
Comment (LudvigHz) [inline app/components/Header/DropdownContents/ProfileDropdown/index.js:40]: I feel like this is weird being a link? and the navlinks above look more like buttons? I think some coherency with the other dropdown menus really won't hurt.

File: app/components/Header/Header.css (hunk 18 above)
Comment (LudvigHz) [reply app/components/Header/Header.css:155]: Yeah, like the other font, this is really, really unnecessary and will just slow the page down for a very slight (worse) font for three words, and you have the extra css of setting the right font in the dropdown containers as well.

Please, just no...

Comment (LudvigHz) [reply app/components/Header/Header.css:155]: Yeah, like the other font, this is really, really unnecessary and will just slow the page down for a very slight (worse) font for three words, and you have the extra css of setting the right font in the dropdown containers as well.

Please, just no...

File: app/components/Header/Header.css [hunk 31]
```diff
@@ -113,70 +122,84 @@
...
 
-  &:hover::after,
-  &.activeItem::after {
-    transform: scaleX(1);
-    bottom: 3px;
-  }
+.navbarItemTitle:hover,
+.activeNavbarItemTitle {
+  opacity: 0.7;
+}
```
Comment (LudvigHz) [inline app/components/Header/Header.css:170]: WIth all the navbars having the same hover style, I find it really weird when hovering over the one with no dropdown (karriere). Since it looks just the same, I kind of expect it to show a menu and just wait for it for a second before realizing there is none. Maybe we could do something to inform the user that it's just a plain link and won't show a dropdown? 

This isn't a priority though, just a little picky thing I noticed.

File: server/pageRenderer.js [hunk 32]
```diff
@@ -112,8 +112,7 @@ export default function pageRenderer({
 
         <link href="https://cdn.jsdelivr.net/npm/font-awesome@4.7.0/css/font-awesome.min.css" rel="stylesheet">
         <link href="https://unpkg.com/ionicons@3.0.0/dist/css/ionicons.min.css" rel="stylesheet">
-        <link href="https://fonts.googleapis.com/css?family=Open+Sans:700|Raleway|Roboto" rel="stylesheet">
-        ${links}
```
Comment (LudvigHz) [inline server/pageRenderer.js:116]: Why remove this?

**Review by LudvigHz — COMMENTED** [review]
This is absolutely a huge improvement, and it looks outstanding. I just have some comments, on the design as well as the code.

**Transitions**. The transition duration of the content in the dropdowns are too long, I get kinda dizzy when cycling through them, and I think speeding them up and decrasing the duration they are visible will really help with this.

**Coherency**. I realize this is a WIP, and I'm guessing this is as well? Since all the dropdowns have different styles, some with a background, some with a caret (this one needs more ligibility for hover styles btw) and some with just icons. Some coherency and more legibility is nice. (I really like the "Om Abakus" one!)

**Logg ut**. All the other bottoms of the dropdown are links, while this is an action button. Don't know exactly what's best here, but some way of differentiating it is nice I think, so that it's obvious it's not a link, but a button performing an action.

**Hamburger menu**. I'm also assuming you have a revamped style for this? Is that going to be part of this PR or another one?

**Comment by LudvigHz** [comment]
And seriously, remove the goddam fonts and use a variant of Roboto or something else that's built-in

File: app/components/Header/DropdownContents/EventsDropdown/EventsDropdown.css (hunk 22 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContents/EventsDropdown/EventsDropdown.css:8]: I totally agree, and the plan is to have everything nicely aligned! I've just been struggling with the design, and needed a small break to gain some creativity 😄

File: app/components/Header/DropdownContents/EventsDropdown/index.js (hunk 23 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContents/EventsDropdown/index.js:28]: I'm not sure what you mean here, and what is this "other"? 🤔 Do you simply not like the use of `<span>`?

File: app/components/Header/DropdownContents/InfoDropdown/index.js (hunk 24 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContents/InfoDropdown/index.js:21]: haha I considered it, and then forgot ... 😅 Will do!

File: app/components/Header/DropdownContents/LoginDropdown.js (hunk 25 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContents/LoginDropdown.js:42]: hmm I wonder what has happened here 🤔 Nice catch! 😍

File: app/components/Header/DropdownContents/LoginDropdown.js (hunk 26 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContents/LoginDropdown.js:51]: You'll have to ask @SmithPeder 🙃 But ye, it's a "dot". I plan on redesigning this Login dropdown anyways, so it will soon cease to exist :)

File: server/pageRenderer.js (hunk 32 above)
Comment (ivarnakken) [reply server/pageRenderer.js:116]: Nothing has been removed, just forgotten to push 🙃 Open Sans behaves a bit weird, as it requires us to specify the weight of the font to be imported (as oppose to Roboto and Raleway)

File: app/components/Header/DropdownContents/NotificationsDropdown/index.js (hunk 27 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContents/NotificationsDropdown/index.js:21]: Nope, but I plan on using it in order for notifications to not be fetched unnecessarily, e.g. when accidentally hovering over the notification icon

File: app/components/Header/DropdownContents/ProfileDropdown/ProfileDropdown.css (hunk 28 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContents/ProfileDropdown/ProfileDropdown.css:1]: Probably not, and I strongly considered it, but thought I would see what you guys thought about it 😅

File: app/components/Header/DropdownContents/ProfileDropdown/ProfileDropdown.css (hunk 29 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContents/ProfileDropdown/ProfileDropdown.css:37]: I've tried to use variables as much as I can, but found some of them a bit lacking. I'm therefore planning on generally improving `variables.css`. The "red" colors there are incorrect (according to (abakus.no/brand)[https://abakus.no/brand], and we also don't need three variables for the same red color. Perhaps I should even ask the PR committee for some consultation regarding what colors are "correct" to use 🤔

File: app/components/Header/Header.css (hunk 31 above)
Comment (ivarnakken) [reply app/components/Header/Header.css:170]: It's a valid point, and something I've thought a lot about. I thought it would be unnecessary to add a unique dropdown to  when not really needed, as it would be more annoying than beneficial. This is also the same approach other websites that utilize a similar navbar have gone for, e.g. [stripe.com](https://stripe.com) where the "Pricing" does not have its own dropdown. I know I've seen others do the same. Now regarding giving the user a different feedback on it ... I'm not sure, as I can see how it would make the link feel more "out of place" 🤔 Thoughts?

File: app/components/Header/DropdownContents/ProfileDropdown/index.js (hunk 30 above)
Comment (ivarnakken) [reply app/components/Header/DropdownContents/ProfileDropdown/index.js:40]: This is also something I've thought a lot about, and thought I would wait and see what you guys thought. Interesting to see that you also caught onto this. Although its not really a "link", it does simultaneously route you to a "new" page, the front page.

File: app/components/Header/Header.css (hunk 31 above)
Comment (ivarnakken) [reply app/components/Header/Header.css:170]: I'll add that stripe actually differentiates the hover effects slightly by changing the cursor, but I like the fact that the navigation links (that people are used to and love) still works as they expect them to and takes them to the same place as before, so that they are not "forced" to utilize the new dropdown. People don't usually like change, and therefore it's essential to not try to fix something that is not broken.

**Comment by juniwbjerde** [comment]
ABA-143

**Comment by juniwbjerde** [comment]
Fixes ABA-143
